ProjectLighthouse.Servers.Website/Controllers/Admin/AdminUserController.cs

```diff
@@ -91,6 +94,64 @@ public AdminUserController(DatabaseContext database)
 
         return this.Redirect($"/user/{targetedUser.UserId}");
     }
+    
+    /// <summary>
+    /// Deletes every comment by the user. Useful in case of mass spam
+    /// </summary>
+    [HttpGet("wipeComments")]
+    public async Task<IActionResult> WipeComments([FromRoute] int id)
+    {
+        UserEntity? user = this.database.UserFromWebRequest(this.Request);
+        if (user == null || !user.IsModerator) return this.NotFound();
+
+        UserEntity? targetedUser = await this.database.Users.FirstOrDefaultAsync(u => u.UserId == id);
+        if (targetedUser == null) return this.NotFound();
+
+        // Get every comment posted by the target user
+        List<CommentEntity> comments = await this.database.Comments
+            .Where(c => c.Poster == targetedUser)
+            .ToListAsync();
+
+        // Loop through and delete all of the comments
+        foreach (CommentEntity comment in comments)
+        {
+            comment.Deleted = true;
+            comment.DeletedBy = user.Username;
+            comment.DeletedType = "moderator";
+            Logger.Success($"Deleted comments for {targetedUser.Username} (id:{targetedUser.UserId})",
```

Comment:
This should probably be outside the loop otherwise it could cause log spam for users with a lot of comments.